src/mmap.rs

```diff
@@ -1196,16 +1196,14 @@ mod tests {
         ];
         let mut iterated_regions = Vec::new();
         let gm = GuestMemoryMmap::from_ranges(&regions).unwrap();
-        let res: guest_memory::Result<()> = gm.with_regions(|_, region| {
```

Comment:
Should we update the unit tests when we actually remove the functions? Even if they're deprecated, they can still be used.

Reply:
Ah, true. I suppose the deprecated functions currently cover the iterator implementation for now too. I'll revert these.